store/sqlstore/channel_store.go

```diff
@@ -2396,6 +2473,34 @@ func (s SqlChannelStore) GetChannelsByIds(channelIds []string, includeDeleted bo
 	return channels, nil
 }
 
+func (s SqlChannelStore) GetChannelsWithTeamDataByIds(channelIds []string, includeDeleted bool) ([]*model.ChannelWithTeamData, error) {
+	keys, params := MapStringsToQueryParams(channelIds, "Channel")
+	query := `SELECT c.*, t.DisplayName AS TeamDisplayName, t.Name AS TeamName, t.UpdateAt AS TeamUpdateAt
+	 FROM Channels c, Teams t
+	 WHERE
+		c.TeamId=t.Id
+		AND
+	  c.Id IN ` + keys + ` ORDER BY c.Name`
+	if !includeDeleted {
+		query = `SELECT c.*, t.DisplayName AS TeamDisplayName, t.Name AS TeamName, t.UpdateAt AS TeamUpdateAt
+			 FROM Channels c, Teams t
+			 WHERE
+				c.TeamId=t.Id
+				AND
+				c.DeleteAt = 0
+				AND
+			  c.Id IN ` + keys + ` ORDER BY c.Name`
+	}
```

Comment:
Could we use squirrel here? Maybe the `!includeDeleted` case would become much simpler.